content/docs/connect/connection-pooling.md

```diff
@@ -49,9 +49,58 @@ The **Connection Details** widget on the Neon **Dashboard** provides **Pooled co
 
 ![Connection Details pooled connection string](/docs/connect/connection_details_pooled.png)
 
+## Optimize queries with PgBouncer and prepared statements
+
+Protocol-level prepared statements are supported with Neon and PgBouncer as of the [PgBouncer 1.21.0 release](https://github.com/pgbouncer/pgbouncer/releases/tag/pgbouncer_1_21_0). Using prepared statements can help boost query performance while providing an added layer of protection against potential SQL injection attacks.
+
+### Understanding prepared statements
+
+A prepared statement in Postgres allows for the optimization of an SQL query by defining its structure once and executing it multiple times with varied parameters. Here's an SQL-level example to illustrate, but please note that direct SQL-level `PREPARE` and `EXECUTE` are not supported with PgBouncer (see [below](#use-prepared-statements-with-pgbouncer)).
+
+```sql
```

Comment:
It seems a bit odd to provide a sample that you can't actually use. Is this because you kind of need to understand the direct SQL-level in order to correctly use the protocol-level? Or the user is familiar with the SQL-level, and so will look in the protocol library for whatever they need?

Maybe modify it to something like this, to guide the user past any weirdness:

"Here's an SQL-level example to illustrate. Note that direct SQL-level `PREPARE` and `EXECUTE` are not supported with PgBouncer (see [below](#use-prepared-statements-with-pgbouncer)), so you can't use this query from the SQL editor. It is meant to give you a clear idea of how a prepared statement works. We'll use the same variables in the protocol-level samples below, so you can see the direct SQL-level translates to different protocols."

And then should we revise the samples to use the same prepared plan name? Maybe "fetch-user" in all cases?

Reply:
Great idea, perhaps for @raoufchebri's blog as well. Modified slightly, as I am not sure I about modifying the Python example code in a way that you might not otherwise.